pkg/devfile/adapters/common/command.go

```diff
@@ -147,29 +147,42 @@ func validateCommandsForGroup(data data.DevfileData, groupType devfilev1.Command
 				defaultCommandCount++
 			}
 		}
-	} else {
+	} else if len(commands) == 1 {
 		// if there is only one command, it is the default command for the group
 		defaultCommandCount = 1
 	}
 
 	if defaultCommandCount == 0 {
-		return fmt.Errorf("there should be exactly one default command for command group %v, currently there is no default command", groupType)
+		return NoDefaultForGroup{Group: groupType}
 	} else if defaultCommandCount > 1 {
-		return fmt.Errorf("there should be exactly one default command for command group %v, currently there is more than one default command", groupType)
+		return MoreDefaultForGroup{Group: groupType}
 	}
 
 	return nil
 }
 
 // GetBuildCommand iterates through the components in the devfile and returns the build command
 func GetBuildCommand(data data.DevfileData, devfileBuildCmd string) (buildCommand devfilev1.Command, err error) {
+	supportedCommand, err := getCommand(data, devfileBuildCmd, devfilev1.BuildCommandGroupKind)
+	// since build command is not necessary
+	// check if we can ignore it
+	if _, ok := err.(NoDefaultForGroup); ok {
```

Comment:
Should there be an `else` block covering the case when error is something other than `NoDefaultForGroup`?

Reply:
Updated the logic